include/revng/Model/IRHelpers.h

```diff
@@ -63,3 +65,30 @@ inline llvm::IntegerType *getLLVMTypeForScalar(llvm::LLVMContext &Context,
   revng_assert(Type.isScalar());
   return getLLVMIntegerTypeFor(Context, Type);
 }
+
+/// Given an LLVM module with a single isolated function, find it and return a
+/// reference to it. This function will assert if there is any other amount (0,
+/// 2 -> inf) number of functions. If the Address is provided it will also
+/// assert that the found function has the expected metaaddress.
+template<typename T>
+  requires(std::same_as<T, llvm::Module> or std::same_as<T, const llvm::Module>)
+decltype(auto)
+getUniqueIsolatedFunction(T &Module, const MetaAddress *Address = nullptr) {
```

Comment:
1. ```
   ConstOrNot<llvm::Module> auto &Module, const MetaAddress *Address = nullptr)
   ```
2. Do you actually use the default argument?
3. In general, I'd avoid sending in a `nullptr`, you can use `MetaAddress::invalid`. For data structures that have an invalid state, it's better to use it instead of `nullptr` or `std::nullopt`.

Reply:
1. Done
2. .
3. Switched to `MetaAddress::invalid()`